source/agora/consensus/Fee.d

```diff
@@ -249,6 +259,11 @@ public class FeeManager
         Amount minimumFee = params.MinFee;
         if (!minimumFee.mul(tx.sizeInBytes()))
             return "Fee: Transaction size overflows fee cap";
+
+        auto freeze_fee = this.getFreezeFee(tx);
+        if (!freeze_fee.isValid() || !minimumFee.add(freeze_fee))
```

Comment:
Since `tx` is user controlled, `getFreezeFee` is allowed to return invalid values.